treee111/wahooMapsCreator
# [BREAKING] X/Y coordinates can be given as input parameters (e.g. for testing!) & refactoring of GUI & CLI processing

## This PR…

- makes it possible to provide X/Y coordinates via command line as input parameter to create the corresponding map

## Considerations and implementations

This PR brings a foundation to implement #100 and #89 by
- refactoring the argparse input coding to be future-proof
  - subparsers (and therefore subcommands) for GUI and CLI start of the tool
  - differentiate between the three possible primary inputs for CLI processing (country-name, X/Y coordinates, json file with tiles) in cli help and program coding

## How to test

1. run tool via CLI with one X/Y coordinate and multiple X/Y coordinates
   - one country involved
   - multiple countries involved
2. run tool via CLI for a country
3. run tool via GUI for a country
4. check README and documentation

## Pull Request Checklist
    + Especially the [Pull-Requests](https://github.com/treee111/wahooMapsCreator/blob/develop/.github/CONTRIBUTING.md#Pull-Requests) section
